### ISSUE TITLE
Issue compiling a gradient descent optimiser using keras

### ISSUE THREAD

**Issue by RasaiStewart** [RasaiStewart/LogisticRegression#1]
Upon using keras to compile a machine learning model that is trained using the mnist dataset from Keras as shown below, I receive the following error: 
![image](https://user-images.githubusercontent.com/60219750/136592521-87d7d108-b0c6-4dfa-b57b-a88780fa8215.png)

# Compile model
optim = keras.optimizers.SGD(lr=0.01)
model.compile(optimizer=optim, loss='categorical_crossentropy', metrics=['accuracy'])
